horace_core/sqw/PixelData/@PixelData/do_binary_op.m

```diff
@@ -43,7 +43,7 @@
     pix_out = binary_op_double_(pix_out, operand, binary_op, flip, npix);
 elseif isa(operand, 'PixelData')
     pix_out = binary_op_pixels_(pix_out, operand, binary_op, flip);
-elseif ~isempty(regexp(class(operand), '^d[0-4]d_old$', 'ONCE')) || isa(operand, 'sigvar')
+elseif ~isempty(regexp(class(operand), '^d[0-4]d_old$', 'ONCE')) || isa(operand, 'DnDBase') || isa(operand, 'sigvar')
     pix_out = binary_op_sigvar_(pix_out, operand, binary_op, flip, npix);
```

Comment:
```suggestion
elseif ~isempty(regexp(class(operand), '^d[0-4]d_old$', 'ONCE')) || isa(operand, 'DnDBase') || isa(operand, 'sigvar')
    pix_out = binary_op_sigvar_(pix_out, operand, binary_op, flip, npix);
 ```
Should it be `^d[0-4]d$' now?
+
Can not judge if we want binary_op on `PixelData` itself rather then doing it as part of operation on `sqw` object. But may be missing something there

Reply:
The `d0d`-`d4d` (new class) case is covered by `DnDBase` check. 

In the `sqw` binary operations implementation, call made as `sqw .op. non_sqw` are implemented as 
```
sqw.pixeldata .op. non_sqw; 
recompute_image;
```
so the logic for the operations is implemented at PixelData level.

You raise a good point about why PixelData needs to know about DnD objects at though - I'll look into this.